src/geometry.cpp

```diff
@@ -208,26 +197,39 @@ find_cell_inner(Particle* p, const NeighborList* neighbor_list)
 
       Lattice& lat {*model::lattices[c.fill_]};
 
+      // Set the position and direction.
+      auto& coord {p->coord_[p->n_coord_]};
+      coord.r = p->r_local();
+      coord.u = p->u_local();
+
+      // Apply translation.
+      coord.r -= c.translation_;
+
+      // Apply rotation.
+      if (!c.rotation_.empty()) {
```

Comment:
An in-place `LocalCoord::rotate()` method could be a way to avoid calling `Position::rotate()` twice and setting `rotated` wherever we want to do this step. You know better than I if that will happen often enough to make it worth the addition.

Reply:
Good suggestion -- I'll add that.